src/sagas/networkSettings.js

```diff
@@ -0,0 +1,179 @@
+import { all, takeEvery, put, call, race, take, delay } from 'redux-saga/effects';
+import { config } from '@hathor/wallet-lib';
+import { isEmpty } from 'lodash';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { featureToggleUpdate, networkSettingsUpdateFailure, networkSettingsUpdateSuccess, reloadWalletRequested, types } from '../actions';
+import { HTTP_REQUEST_TIMEOUT, NETWORK, networkSettingsKey, NETWORK_TESTNET, STAGE, STAGE_DEV_PRIVNET, STAGE_TESTNET } from '../constants';
+import { getFullnodeNetwork, getWalletServiceNetwork } from './helpers';
+import { STORE } from '../store';
+
+export function* updateNetworkSettings(action) {
+  const {
+    nodeUrl,
+    explorerUrl,
+    walletServiceUrl,
+  } = action.payload || {};
+
+  // validates input emptyness
+  if (isEmpty(action.payload)) {
+    yield put(networkSettingsUpdateFailure());
+    return;
+  }
+
+  // validates explorerUrl
+  // - required
+  // - should have a valid URL
+  if (isEmpty(explorerUrl) || invalidUrl(explorerUrl)) {
+    yield put(networkSettingsUpdateFailure());
+    return;
+  }
+
+  // validates nodeUrl
+  // - required
+  // - should have a valid URl
+  if (isEmpty(nodeUrl) || invalidUrl(nodeUrl)) {
+    yield put(networkSettingsUpdateFailure());
+    return;
+  }
+
+  // validates walletServiceUrl
+  // - optional
+  // - should have a valid URL, if given
+  if (walletServiceUrl && invalidUrl(walletServiceUrl)) {
+    yield put(networkSettingsUpdateFailure());
+    return;
+  }
+
+  // NOTE: Should we allow that all the URLs be equals?
+  // In practice they will never be equals.
+
+  const oldExplorerUrl = config.getExplorerServiceBaseUrl();
+  const oldNodeUrl = config.getServerUrl();
+  const oldWalletServiceUrl = config.getWalletServiceBaseUrl();
+  const oldWalletServiceWsUrl = config.getWalletServiceBaseWsUrl();
+
+  config.setExplorerServiceBaseUrl(explorerUrl);
+  config.setServerUrl(nodeUrl);
+
+  // TODO: get network as the best effort
+  // - walletServiceUrl has precedence
+  // - nodeUrl as fallback
+
+  let network;
+  let walletServiceWsUrl;
+  if (walletServiceUrl) {
+    const _url = new URL(walletServiceUrl);
+    if (_url.protocol.includes('https')) {
```

Comment:
The websocket endpoint might have a different URL as it's a different lambda, so this might not work for everyone, specially during development (which is our current usecase, right?)

Reply:
Implemented at e28fe89037a750a5958280be3099aded816220d3